test/classes/initializers/generics/phase1/nested-function-mods-field1.bad

```diff
@@ -1,6 +1,6 @@
 nested-function-mods-field1.chpl:6: In initializer:
 nested-function-mods-field1.chpl:10: error: illegal lvalue in assignment
-  nested-function-mods-field1.chpl:16: called as borrowed Foo.init(param val = 13)
+  nested-function-mods-field1.chpl:16: called as (borrowed Foo(13)).init(param val = 13)
```

Comment:
This one's interesting. Would it be easy to squash the type's `13`? I find this unnecessarily complex, and wish we can have the old version of the error message (am fine with parentheses). I think my concern is only applicable to `init` calls.

Reply:
I can look at simplifying it for `init` calls for now. (I could see an argument in some cases where we might wish to see the type being initialized separate from the arguments but we can leave that as future work).

Reply:
We might want to revisit this choice when we handle type aliases and typeless fields better -- see https://github.com/chapel-lang/chapel/issues/16508#issuecomment-721270436

Reply:
I'm going to make it print `called as Foo.init(param val = 13)` since the `borrowed` part doesn't really make sense either in the context of initialization.

Reply:
(I went further and removed the `borrowed` part from `this` arguments for class method calls since it is the normal case and this way it better mirrors declaration syntax).